inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp

```diff
@@ -247,15 +253,58 @@ static void Transformation(ICNNNetwork::Ptr& clonedNetwork, const Config& conf)
         return false;
     });
 
-    legacyManager.get_pass_config()->set_callback<ngraph::pass::UnrollTensorIterator>([](const_node_ptr &node) -> bool {
+    legacyPassConfig->set_callback<ngraph::pass::UnrollTensorIterator>([](const_node_ptr &node) -> bool {
         // UnrollTI transformation is disabled by default, is turned on by LowLatency transformation
         return node->get_rt_info().count("UNROLL_TI") == 0;
     });
+
+    auto initAxisIdx = [](const std::shared_ptr<ngraph::Node> node) -> int {
+        int axisIdx = 0, numberOfNonUnit = 0;
+
+        for (size_t i = 0; i < node->get_shape().size(); i++) {
+            if (node->get_shape()[i] > 1) {
+                axisIdx = i;
+                numberOfNonUnit++;
+            }
+        }
+        return numberOfNonUnit > 1 ? -1 : axisIdx;
+    };
+    auto isSupportedFQ = [initAxisIdx](const_node_ptr &node) {
+        std::set<int> quantizationParamsAxisesIdxs;
+        std::set<size_t> quantizationParamsAxisesSizes;
+        for (size_t i = 1; i < node->get_input_size(); i++) {
+            auto inNode = node->get_input_node_shared_ptr(i);
+            auto axis = initAxisIdx(inNode);
+            if (axis == -1)
+                return false;
+            if (inNode->get_shape().size() != 0 && inNode->get_shape()[axis] != 1) {
+                quantizationParamsAxisesIdxs.insert(axis);
+                quantizationParamsAxisesSizes.insert(inNode->get_shape()[axis]);
+            }
+        }
+        return (quantizationParamsAxisesIdxs.size() <= 1 && quantizationParamsAxisesSizes.size() <= 1);
+    };
+
+    legacyPassConfig->set_callback<ngraph::pass::FakeQuantizeDecomposition>([isSupportedFQ](const_node_ptr &node) -> bool {
+        if (auto fq_op = std::dynamic_pointer_cast<const ngraph::opset1::FakeQuantize>(node)) {
+            if (node->get_input_node_shared_ptr(0)->get_shape().size() > 5)
+                return false;
+            for (size_t i = 1; i < fq_op->get_input_size(); i++) {
+                if (!std::dynamic_pointer_cast<const ngraph::opset1::Constant>(fq_op->get_input_node_shared_ptr(i)) ||
+                    node->get_input_node_shared_ptr(i)->get_shape().size() > 5)
+                    return false;
+            }
+            return isSupportedFQ(fq_op);
+        }
+
+        return true;
+    });
+
     legacyManager.run_passes(nGraphFunc);
 
     OV_ITT_TASK_CHAIN(taskChain, MKLDNNPlugin::itt::domains::MKLDNN_LT, "Transformation", "convertFunctionToICNNNetwork");
 
-    clonedNetwork = InferenceEngine::details::convertFunctionToICNNNetwork(nGraphFunc, *clonedNetwork);
+    clonedNetwork = InferenceEngine::details::convertFunctionToICNNNetwork(nGraphFunc, *clonedNetwork, keep_constant_inputs);
```

Comment:
In ```convertFunctionToICNNNetwork``` we have check: if model contain FakeQuantize we set ```keep_constants = true``` https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/legacy_api/src/convert_function_to_cnn_network.cpp#L1735
and for all ```ConvolutionIE``` constant will be keep like a node https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/legacy_api/src/convert_function_to_cnn_network.cpp#L1635

so this check does not change previous behaviour, we just extract it from ```convertFunctionToICNNNetwork``` 
this is neccesary to do because all FakeQuantize could be decomposed and internal check in ```convertFunctionToICNNNetwork``` will be set ```keep_constants = false```
and this change previous behaviour